akd/src/directory.rs

```diff
@@ -24,7 +24,7 @@ use log::{debug, error, info};
 use rand::{distributions::Alphanumeric, CryptoRng, Rng};
 
 use std::collections::HashMap;
-use std::marker::{Send, Sync};
+use std::marker::{PhantomData, Send, Sync};
```

Comment:
Question: Is there a reason we're putting phantomdata back? We did a big removal at one point to remove it, but I guess this maybe makes a cleaner API?

Reply:
Oh, oops, what was the reason for removing PhantomData? I was indeed thinking that this makes for a cleaner API but there are alternatives if we want to avoid using PhantomData.

Reply:
I think it was to minimize the number of generics on the struct, but it's also isn't a big "technical" change. Probably not specifying the generic on every method call is overall easier I guess.